server/datastore/mysql/hosts.go

```diff
@@ -622,17 +622,27 @@ func (ds *Datastore) applyHostFilters(opt fleet.HostListOptions, sql string, fil
 		softwareFilter, munkiFilter, lowDiskSpaceFilter,
 	)
 
-	sql, params = filterHostsByStatus(sql, opt, params)
+	now := ds.clock.Now()
+	sql, params = filterHostsByStatus(now, sql, opt, params)
 	sql, params = filterHostsByTeam(sql, opt, params)
 	sql, params = filterHostsByPolicy(sql, opt, params)
 	sql, params = filterHostsByMDM(sql, opt, params)
 	sql, params = filterHostsByOS(sql, opt, params)
+	sql, params = filterHostsByMissing(now, sql, opt, params)
 	sql, params = hostSearchLike(sql, params, opt.MatchQuery, hostSearchColumns...)
 	sql, params = appendListOptionsWithCursorToSQL(sql, params, opt.ListOptions)
 
 	return sql, params
 }
 
+func filterHostsByMissing(now time.Time, sql string, opt fleet.HostListOptions, params []interface{}) (string, []interface{}) {
```

Comment:
question: given that this filter will be part of the `filterHostByStatus` "family", does it make sense to move this code to be a switch case in that function?

<img width="688" alt="image" src="https://user-images.githubusercontent.com/4419992/191980027-ccd21848-bdec-4e5f-839b-d0bdbeba52e9.png">

this also has the advantage that you can't filter by `new` and `missing_10_days` at the same time, unless you're trying to avoid exactly that!